sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_eth.c

```diff
@@ -0,0 +1,195 @@
+#include "net/gnrc.h"
+#include "net/gnrc/gnrc_netdev2.h"
+#include "net/ethernet/hdr.h"
+
+#include "od.h"
+
+#define ENABLE_DEBUG (0)
+#include "debug.h"
+
+static gnrc_pktsnip_t *_recv(gnrc_netdev2_t *gnrc_netdev2)
```

Comment:
> I think you're mixing up hdr and eth_hdr in the last lines, right?

Right, that tends to happen when you head-compile ^^

> Where do you mark the payload according to ethertype?

Forgot it. Just throw a

```C
pkt->type = gnrc_nettype_from_ethertype(byteorder_ntohs(hdr->type));
```

in there after you set `hdr` (the real one this time), but before you remove `eth_hdr` in the `gnrc_pktbuf_remove_snip(pkt, hdr);` line.